src/devtools/client/debugger/src/components/App.js

```diff
@@ -284,6 +285,20 @@ Debugger.childContextTypes = {
   l10n: PropTypes.object,
 };
 
+function DebuggerWaitingForEditor(props) {
+  const [hasEditor, setHasEditor] = useState(false);
+
+  useEffect(() => {
+    async function awaitEditor() {
```

Comment:
style thing, but an iife would be cleaner. and perhaps `setLoading` instead of editor, we might want to lode other things as well.

```js
useEffect(() => {
  (async () => {
    await waitForEditor();
    setLoading(false)
  })()
}, [])
```

Reply:
Actually I think an IIFE is less readable. But I already noticed this is a minority opinion among javascript developers.

Reply:
hehe